conans/client/generators/gcc.py

```diff
@@ -12,17 +15,39 @@ def content(self):
         """With gcc_flags you can invoke gcc like that:
         $ gcc main.c @conanbuildinfo.gcc -o main
         """
-        defines = " ".join("-D%s" % x for x in self._deps_build_info.defines)
-        include_paths = " ".join("-I%s"
-                                 % x.replace("\\", "/") for x in self._deps_build_info.include_paths)
-        rpaths = " ".join("-Wl,-rpath=%s"
-                          % x.replace("\\", "/") for x in self._deps_build_info.lib_paths)
-        lib_paths = " ".join("-L%s" % x.replace("\\", "/") for x in self._deps_build_info.lib_paths)
-        libs = " ".join("-l%s" % x for x in self._deps_build_info.libs)
-        other_flags = " ".join(self._deps_build_info.cppflags +
-                               self._deps_build_info.cflags +
-                               self._deps_build_info.sharedlinkflags +
-                               self._deps_build_info.exelinkflags)
-        flags = ("%s %s %s %s %s %s"
-                 % (defines, include_paths, lib_paths, rpaths, libs, other_flags))
-        return flags
+        flags = []
+        flags.extend(["-D%s" % x for x in self._deps_build_info.defines])
+        flags.extend(['-I"%s"' % x.replace("\\", "/") for x in self._deps_build_info.include_paths])
+        rpath_separator = "," if platform.system() == "Darwin" else "="
+        flags.extend(['-Wl,-rpath%s"%s"' % (rpath_separator, x.replace("\\", "/")) 
+                      for x in self._deps_build_info.lib_paths]) # rpaths
+        flags.extend(['-L"%s"' % x.replace("\\", "/") for x in self._deps_build_info.lib_paths])
+        flags.extend(["-l%s" % x for x in self._deps_build_info.libs])
+        flags.extend(self._deps_build_info.cppflags)
+        flags.extend(self._deps_build_info.cflags)
+        flags.extend(self._deps_build_info.sharedlinkflags)
+        flags.extend(self._deps_build_info.exelinkflags)
+        flags.extend(self._libcxx_flags())
```

Comment:
Seems some duplication with AutoToolsEnvironment, in _libcxx_flags and architecture, please check if makes sense to refactor.